### ISSUE TITLE
`Style/MethodCallWithArgsParentheses` fails to detect violations in test blocks using `test "name" do` syntax

### ISSUE THREAD

**Issue by mmenanno** [rubocop/rubocop#14569]
## Expected behavior

The `Style/MethodCallWithArgsParentheses` cop should detect method calls without parentheses consistently across all code contexts, including within `test "name" do` blocks when using test frameworks like Minitest that support this syntax.

## Actual behavior

The cop only detects violations in traditional `def test_` method definitions but fails to detect identical violations within `test "name" do` block syntax, even though both contexts contain the same method call pattern.

## Steps to reproduce the problem

1. Create a test file with both `def test_` and `test "name" do` formats:

```ruby
# frozen_string_literal: true

require "minitest/autorun"

class MethodCallParenthesesTest < Minitest::Test
  def test_with_def_style
    assert_equal(1, 1)  # This has parentheses
    assert_equal 1, 1   # This should be flagged - AND IS
  end

  test "with do/end block" do
    assert_equal(1, 1)  # This has parentheses
    assert_equal 1, 1   # This should be flagged - BUT IS NOT
  end
end
```

2. Create a `.rubocop.yml` file to isolate the cop:

```yaml
AllCops:
  DisabledByDefault: true

Style/MethodCallWithArgsParentheses:
  Enabled: true
  EnforcedStyle: require_parentheses
```

3. Run RuboCop with debug output:

```bash
$ rubocop --debug
For /Users/[username]/dev/rubocop_issue_test: configuration from /Users/[username]/dev/rubocop_issue_test/.rubocop.yml
Default configuration from /Users/[username]/.local/share/mise/installs/ruby/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.81.1/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/[username]/dev/rubocop_issue_test/method_call__with_args_test.rb
Loading cache from /Users/[username]/.cache/rubocop_cache/8b47d778c718e8abc8048a1983ecf9b7e3d13d26/6d7a3b621ca1730e04accd938619e4bdab66cfb1/3fd22c96bcb5437a8b641bca8c6cf756b628c387
C

Offenses:

method_call__with_args_test.rb:8:5: C: [Correctable] Style/MethodCallWithArgsParentheses: Use parentheses for method calls with arguments.
    assert_equal 1, 1
    ^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
Finished in 0.05305800004862249 seconds
```

**Expected:** Both `assert_equal 1, 1` calls (lines 8 and 13) should be flagged.
**Actual:** Only the one in the `def test_` method is flagged; the identical call in the `test "name" do` block is ignored.

## RuboCop version

```bash
$ rubocop -V
1.81.1 (using Parser 3.3.9.0, Prism 1.5.1, rubocop-ast 1.47.1, analyzing as Ruby 3.4, running on ruby 3.4.5) [arm64-darwin24]
```

**Comment by mmenanno** [comment]
I think this is partially intentional since the test block is considered a macro. If I put every single assert / refute method into IncludedMacros it would work around this, but that would be a pain to maintain. I've put up a PR that you can see int he timeline above that adds an IncludedMacroPatterns config option to fix this / better work around it.

Another potential fix could be to create an active support specific extension fix to make the contents of test blocks not be affected by this.
